action/recipes/recipes.go

```diff
@@ -74,7 +74,10 @@ func commonGetOpts(get getValFunc) (commonOpts, error) {
 	}
 
 	// Check if sdk_version is URL, if not make it into URL defaulting to our containers
-	if _, err := url.ParseRequestURI(opts.sdkVersion); err != nil {
+	//   WARNING: for url.ParseRequestURI is "edk2-stable202105:main" a valid URL for whatever reason ...
```

Comment:
The URI parsing follows the RFC 3986 and technically the mentioned string falls into that scheme even though it makes no sense in the context of using this as an FQDN. So while the string is a valid URI it is not a valid FQDN which you'll try to achieve. Though I think this approach here is good enough to also match for valid FQDNs.

Reply:
I see